src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,229 @@
+//! This module provides a view for breadcrumbs, enabling us to know which node the graph being
+//! edited belongs to.
+
+use crate::prelude::*;
+
+pub mod breadcrumb;
+pub use breadcrumb::Breadcrumb;
+
+use enso_frp as frp;
+use ensogl::display;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+
+
+
+// =================
+// === FrpInputs ===
+// =================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpInputs {
+    /// Push breadcrumb.
+    pub push_breadcrumb : frp::Source<String>,
+    /// Pop breadcrumb.
+    pub pop_breadcrumb : frp::Source,
+}
+
+impl FrpInputs {
+    /// Create new FrpInputs.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            def push_breadcrumb = source();
+            def pop_breadcrumb  = source();
+        }
+        Self{push_breadcrumb,pop_breadcrumb}
+    }
+}
+
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpOutputs {
+    pub pushed_breadcrumb : frp::Source<String>,
+    pub popped_breadcrumb : frp::Source<String>
```

Comment:
do not use past perfect in names - frp names describe something that can happen both now as something that happened in the past. So just use names like `push_breadcrumb` instead